parse.go

```diff
@@ -134,6 +136,19 @@ func ParseConfigMap(r io.Reader) (*v1.ConfigMap, error) {
 	return &cm, nil
 }
 
+// ParseCustomResourceDefinition parses a Custom Resource Definition
+func ParseCustomResourceDefinition(r io.Reader) (*extensionsv1beta1.CustomResourceDefinition, error) {
+	if r == nil {
+		return nil, trace.BadParameter("missing reader")
```

Comment:
nit: I think we can assume this is a required parameter as it is the only parameter so I'd remove this. Otherwise it just clutters.